cardano-testnet/src/Testnet/Util/Runtime.hs

```diff
@@ -117,22 +122,38 @@ readNodeLoggingFormat = \case
 allNodes :: TestnetRuntime -> [NodeRuntime]
 allNodes tr = bftNodes tr <> fmap poolRuntime (poolNodes tr)
 
+makeSprocket
+  :: TmpPath
+  -> String
+  -> Sprocket
+makeSprocket tmpPath node
+  = Sprocket (getTmpBaseAbsPath tmpPath)  (getSocketDir tmpPath </> node)
+newtype TmpPath = TmpPath
+  { unTmpPath :: FilePath
+  } deriving (Eq, Show)
+getTmpRelPath :: TmpPath -> FilePath
+getTmpRelPath (TmpPath fp) = FP.makeRelative (getTmpBaseAbsPath (TmpPath fp)) fp
+getSocketDir :: TmpPath -> FilePath
+getSocketDir fp = getTmpRelPath fp </> "socket"
+getTmpBaseAbsPath :: TmpPath -> FilePath
+getTmpBaseAbsPath (TmpPath fp) = FP.takeDirectory fp
+getLogDir :: TmpPath -> FilePath
+getLogDir (TmpPath fp) = fp </> "logs"
```

Comment:
Looks a bit crowded.  Can you spread them out with newlines?

Reply:
ups yes some blank lines got messed up during the rebase